File: isegoria/src/main/java/com/eulersbridge/isegoria/models/Election.java
package com.eulersbridge.isegoria.models;

import com.eulersbridge.isegoria.network.Timestamp;
import com.squareup.moshi.Json;

@SuppressWarnings("unused")
public class Election {

    @Json(name = "electionId")
    public long id;

    @Json(name = "start")
    @Timestamp
    public long startTimestamp;

    @Json(name = "end")
    @Timestamp
    public long endTimestamp;

    @Json(name = "startVoting")
    @Timestamp
    public long startVotingTimestamp;

    @Json(name = "endVoting")
    @Timestamp
    public long endVotingTimestamp;

    public String title;
    public String introduction;
    public String process;

    @Json(name = "institutionDomain")
    private ElectionInstitution electionInstitution;
}

File: isegoria/src/main/java/com/eulersbridge/isegoria/network/api/models/Election.kt
package com.eulersbridge.isegoria.network.api.models

import com.eulersbridge.isegoria.network.adapters.Timestamp
import com.squareup.moshi.Json

data class Election (
    @Json(name = "electionId")
    val id: Long,

    @Timestamp
    val start: Long,

    @Timestamp
    val end: Long,

    @Timestamp
    val startVoting: Long,

    @Timestamp
    val endVoting: Long,

    val title: String?,
    val introduction: String?,
    val process: String?,

    @Json(name = "institutionDomain")
    private val electionInstitution: ElectionInstitution? = null
)